packages/dd-trace/src/appsec/iast/index.js

```diff
@@ -105,4 +114,13 @@ function onIncomingHttpRequestEnd (data) {
   }
 }
 
+// Response headers are collected here because they are not available in the onIncomingHttpRequestEnd when using Fastify
+function onResponseWriteHeadCollect ({ res, responseHeaders = {} }) {
```

Comment:
Do responseHeaders keys come always in lower case?

When [retrieving headers in analyzers](https://github.com/DataDog/dd-trace-js/pull/6072/files#diff-7898738aed765e3d9dafa59774bec49e7fbd2052ea0714d2470857849136df54R32) the code searches by header name in lower case. If the header name from fastify does not come in lower case, they must be lower cased before setting them in the weak map.

Reply:
I've checked they come in lower case. I'm resolving this conversation.